daemon/container_unix.go

```diff
@@ -938,22 +940,31 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 	}
 
 	ep, err := container.getEndpointInNetwork(n)
-	if err != nil {
-		if _, ok := err.(libnetwork.ErrNoSuchEndpoint); !ok {
-			return err
-		}
+	if err == nil {
```

Comment:
@cpuguy83 This is protect and give a meaningful error message to a user who is trying to connect the container to a network with which it is already connected